zeebe/broker/src/main/java/io/camunda/zeebe/broker/transport/snapshotapi/SnapshotApiRequestHandler.java

```diff
@@ -91,15 +91,24 @@ protected ActorFuture<Either<ErrorResponseWriter, SnapshotApiResponseWriter>> ha
                   LOG.atLevel(Level.DEBUG)
                       .addKeyValue("transferId", request.transferId())
                       .log("Last processed position is {}", lastProcessedPosition);
-                  return service
-                      .getLatestSnapshot(partitionId, lastProcessedPosition, request.transferId())
-                      .thenApply(
-                          chunk -> {
-                            responseWriter.setResponse(
-                                new SnapshotChunkResponse(
-                                    request.transferId(), Optional.ofNullable(chunk)));
-                            return Either.right(responseWriter);
-                          });
+                  return service.getLatestSnapshot(
+                      partitionId, lastProcessedPosition, request.transferId());
+                },
+                actor)
+            .andThen(
+                (chunk, error) -> {
+                  if (error != null) {
+                    LOG.error(
+                        "Failed to get the latest snapshot for partition {}: {}",
```

Comment:
:upside_down_face: Partition id should be in the context already and just giving the exception should be enough, no need to write the error message again.